test/unit/StakingPool/helpers.js

```diff
@@ -29,6 +29,61 @@ const setTime = async timestamp => {
   await mineNextBlock();
 };
 
+async function calculateBasePrice(stakingPool, product, priceChangePerDay) {
+  const { timestamp } = await ethers.provider.getBlock('latest');
+  const timeSinceLastUpdate = BigNumber.from(timestamp).sub(product.nextPriceUpdateTime);
+  const priceDrop = timeSinceLastUpdate.mul(priceChangePerDay).div(daysToSeconds('1'));
+  const basePrice = product.nextPrice.lt(product.targetPrice.add(priceDrop))
+    ? product.targetPrice
+    : product.nextPrice.sub(priceDrop);
+  return basePrice;
+}
+
+async function calculateSurgePremium(stakingPool, coverAmount, initialCapacityUsed, totalCapacity, config) {
+  const amountOnSurge = calculateAmountOnSurge(coverAmount, initialCapacityUsed, totalCapacity, config);
+  const surgePremium = await stakingPool.calculateSurgePremium(amountOnSurge, totalCapacity);
+  const amountOnSurgeSkipped = calculateAmountOnSurgeSkipped(coverAmount, initialCapacityUsed, totalCapacity, config);
+  const surgePremiumSkipped = await stakingPool.calculateSurgePremium(amountOnSurgeSkipped, totalCapacity);
+  return { surgePremium, surgePremiumSkipped };
+}
+
+// config is from StakingPool/unit/setup.js
+function calculateAmountOnSurge(coverAmount, initialCapacityUsed, totalCapacity, config) {
+  coverAmount = BigNumber.from(coverAmount).div(config.NXM_PER_ALLOCATION_UNIT);
+  initialCapacityUsed = BigNumber.from(initialCapacityUsed);
+  totalCapacity = BigNumber.from(totalCapacity);
+  const finalCapacityUsed = initialCapacityUsed.add(coverAmount);
+  const surgeStartPoint = totalCapacity.mul(config.SURGE_THRESHOLD_RATIO) / config.SURGE_THRESHOLD_DENOMINATOR;
+  const amountOnSurge = finalCapacityUsed.sub(surgeStartPoint);
+  return amountOnSurge;
+}
+
+// This function should calculate the amount on surge skipped
+function calculateAmountOnSurgeSkipped(coverAmount, initialCapacityUsed, totalCapacity, config) {
+  coverAmount = BigNumber.from(coverAmount).div(config.NXM_PER_ALLOCATION_UNIT);
+  initialCapacityUsed = BigNumber.from(initialCapacityUsed);
+  totalCapacity = BigNumber.from(totalCapacity);
+  const surgeStartPoint = totalCapacity.mul(config.SURGE_THRESHOLD_RATIO).div(config.SURGE_THRESHOLD_DENOMINATOR);
+
+  const finalCapacityUsed = initialCapacityUsed.add(coverAmount);
+
+  if (finalCapacityUsed.lte(surgeStartPoint)) {
+    return BigNumber.from(0);
+  }
+
+  if (initialCapacityUsed.lte(surgeStartPoint)) {
+    return BigNumber.from(0);
+  }
+
+  return initialCapacityUsed.sub(surgeStartPoint);
+}
+
+// Note fn expects coverAmount is rounded down by NXM_PER_ALLOCATION_UNIT
```

Comment:
Comment seems wrong. `NXM_PER_ALLOCATION_UNIT` is the smallest unit of allocation the staking pool works with. Its value is 0.01 nxm. If anyone wants to allocate less than that - it is rounded UP to it (not down).